config/config.go

```diff
@@ -138,25 +113,31 @@ func Parse(reader io.Reader, location string) (conf Config, err error) {
 	return conf, err
 }
 
+// replaceEnvVars replaces environment variable placeholders in reader stream with values
+// i.e. "val = $VAR" -> "val = 3"
 func replaceEnvVars(reader io.Reader) (io.Reader, error) {
-	// Replaces environment variable placeholders in reader stream with values
-	// i.e. "val = $VAR" -> "val = 3"
 	// Variable definition follows IEEE Std 1003.1-2001
 	//   A dollar sign ($) followed by an upper-case letter, followed by
 	//   zero or more upper-case letters, digits, or underscores (_).
 	regexStr := `\$[A-Z]+[A-Z1-9_]*`
 	varFinder := regexp.MustCompile(regexStr)
 	configBytes, err := ioutil.ReadAll(reader)
 	if err != nil {
-		log.Printf("Problem reading from config reader: %v", err)
 		return nil, err
 	}
+
 	configStr := string(configBytes)
 
 	varPlaceHolders := varFinder.FindAllString(configStr, -1)
 	for _, ph := range varPlaceHolders {
 		// Get the environment variable value (drop the leading dollar sign ($))
 		envVal := os.Getenv(ph[1:])
```

Comment:
Use `syscall.Getenv()` instead. It will tell you whether the variable exists or not.

Reply:
Good call. fixing coming up